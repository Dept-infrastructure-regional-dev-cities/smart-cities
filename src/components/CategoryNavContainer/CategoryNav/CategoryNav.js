import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import style from './CategoryNav.scss';
import { NO_CATEGORY } from '../../../constants';

const CategoryNav = props => (
  <div>
    <NavLink
      className={style.link}
      activeClassName={style.link__active}
      to={`/${props.cityId}/${NO_CATEGORY}`}
    >Overview</NavLink>

    {props.categories.map(category => (
      <NavLink
        className={style.link}
        activeClassName={style.link__active}
        key={category.id}
        to={`/${props.cityId}/${category.id}`}
      > {category.name} </NavLink>
    ))}
  </div>
);

CategoryNav.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  })).isRequired,
  cityId: PropTypes.string.isRequired,
};

export default CategoryNav;
