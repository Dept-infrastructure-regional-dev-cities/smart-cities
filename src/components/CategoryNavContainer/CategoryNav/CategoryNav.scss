@import '~style/variables';

.link {
  display: inline-block;
  padding: 10px;
  text-decoration: none;
}

.link__active {
  background: grey;
  color: white;
}
