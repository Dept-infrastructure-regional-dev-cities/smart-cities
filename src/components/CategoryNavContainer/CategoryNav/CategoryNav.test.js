import React from 'react';
import { shallow } from 'enzyme';
import CategoryNav from './CategoryNav';

const defaultProps = {
  cityId: 'mock',
  categories: [
    {
      name: 'Perth',
      id: 'perth',
    },
    {
      name: 'Sydney',
      id: 'sydney',
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <CategoryNav {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
