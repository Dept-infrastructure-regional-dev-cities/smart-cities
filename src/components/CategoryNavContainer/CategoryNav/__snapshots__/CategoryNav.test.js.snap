// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div>
  <NavLink className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/mock/overview\\" ariaCurrent=\\"true\\">
    Overview
  </NavLink>
  <NavLink className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/mock/perth\\" ariaCurrent=\\"true\\">
     
    Perth
     
  </NavLink>
  <NavLink className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/mock/sydney\\" ariaCurrent=\\"true\\">
     
    Sydney
     
  </NavLink>
</div>"
`;
